src/prefect/server/api/deployments.py

```diff
@@ -318,6 +318,10 @@ async def get_scheduled_flow_runs_for_deployments(
             sort=schemas.sorting.FlowRunSort.NEXT_SCHEDULED_START_TIME_ASC,
         )
 
+        await models.deployments._update_deployment_last_polled(
+            session=session, deployment_ids=deployment_ids
+        )
+
```

Comment:
I think it makes sense to first ensure we are able to retrieve all flow runs successfully before updating the timestamp on `deployment.last_polled`. However, since we return the `FlowRunResponse` inside the session context, I am doing a bulk update on `deployment.last_polled` in the same session. Would appreciate thoughts on whether it might be better to move to a different session or to move the response return outside of the session.

Reply:
I actually just went with the latter so that I could lock the deployment rows during the bulk update, but again would appreciate thoughts on it.

Reply:
If I remember correctly, the response is inside the session context because `FlowRunResponse.from_orm` may need additional database calls to retrieve the work pool and work queue names. The way that SQLAlchemy loads data can remove the need for those additional database calls. The tests pass, so I think moving it out of the session context is okay, but it might be worth checking `models.flow_runs.read_flow_runs` to be sure.

Reply:
Thanks, I'll take a look. It might just be easiest to pass the responses into a variable instead of returning there.

Reply:
Went with what I mentioned above, applied in [4a4ef0b](https://github.com/PrefectHQ/prefect/pull/10969/commits/4a4ef0b2d25f02c0424c0986ca90e33c450fb5cd)